flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/LocalInputPreferredSlotSharingStrategyTest.java

```diff
@@ -301,6 +302,57 @@ public void testInputLocalityIsRespectedWithTwoEdgesBetweenTwoVertices() throws
         }
     }
 
+    @Test
+    public void testGetExecutionSlotSharingGroupOfLateAttachedVertices() {
+
+        JobVertexID jobVertexID1 = new JobVertexID();
+        JobVertexID jobVertexID2 = new JobVertexID();
+        JobVertexID jobVertexID3 = new JobVertexID();
+
+        final SlotSharingGroup slotSharingGroup1 = new SlotSharingGroup();
+        slotSharingGroup1.addVertexToGroup(jobVertexID1);
+        slotSharingGroup1.addVertexToGroup(jobVertexID2);
+
+        final SlotSharingGroup slotSharingGroup2 = new SlotSharingGroup();
+        slotSharingGroup2.addVertexToGroup(jobVertexID3);
+
+        TestingSchedulingTopology topology = new TestingSchedulingTopology();
+
+        TestingSchedulingExecutionVertex ev1 = topology.newExecutionVertex(jobVertexID1, 0);
+        TestingSchedulingExecutionVertex ev2 = topology.newExecutionVertex(jobVertexID2, 0);
+        topology.connect(ev1, ev2);
+
+        final LocalInputPreferredSlotSharingStrategy strategy =
+                new LocalInputPreferredSlotSharingStrategy(
+                        topology,
+                        new HashSet<>(Arrays.asList(slotSharingGroup1, slotSharingGroup2)),
+                        Collections.emptySet());
+
+        assertThat(strategy.getExecutionSlotSharingGroups().size(), is(1));
+        assertThat(
+                strategy.getExecutionSlotSharingGroup(ev1.getId()).getExecutionVertexIds(),
+                containsInAnyOrder(ev1.getId(), ev2.getId()));
+        assertThat(
+                strategy.getExecutionSlotSharingGroup(ev2.getId()).getExecutionVertexIds(),
+                containsInAnyOrder(ev1.getId(), ev2.getId()));
+
+        // add new job vertices and notify scheduling topology updated
+        TestingSchedulingExecutionVertex ev3 = topology.newExecutionVertex(jobVertexID3, 0);
+        topology.connect(ev2, ev3);
```

Comment:
-> topology.connect(ev2, ev3, ResultPartitionType.BLOCKING);

a.t.m we do not expect pipelined downstream vertices to be attached dynamically